src/Raven.Server/Documents/PeriodicBackup/DirectUpload/MultipleFileUploaderBase.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.Backups;
+using Raven.Server.Documents.PeriodicBackup.Retention;
+using Raven.Server.ServerWide;
+using Sparrow.Collections;
+using Sparrow.Server.Logging;
+
+namespace Raven.Server.Documents.PeriodicBackup.DirectUpload;
+
+public abstract class MultipleFileUploaderBase<T> : DirectFileUploader
+{
+    protected readonly LinkedList<T> _threads;
+    protected readonly ConcurrentSet<Exception> _exceptions;
```

Comment:
It isn't threads.
Why is this generic - who else is using that?
What is there both `ConcurrentSet` and `LinkedList` (not thread safe)